### ISSUE TITLE
Embeds Blocks don't differentiate a preview error from a oEmbed API error

### ISSUE THREAD

**Issue by 7studio** [WordPress/gutenberg#8361]
**Describe the bug**
When I try to use the Embed Blocks, the preview of a valid embeddable content and the error message for an invalid URL have the same render.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a new content.
2. Add a Facebook Embed Block.
3. Paste a valid content URL. E.g.: https://www.facebook.com/oohmyworld/videos/2324195200940031/
4. See the error message "Previews for this are unavailable in the editor, sorry!".
5. Add a Facebook Embed Block.
6. Paste an invalid content URL. E.g.: https://www.facebook.com/oohmyworld/vides/2324195200940031/
7. See the error message "Previews for this are unavailable in the editor, sorry!".

**Expected behaviour**
Gutenberg should display different error messages for failed previews and failed oEmbed API requests.
E.g.:
- Failed preview: "Previews for this are unavailable in the editor, sorry!"
- oEmbed API error: "This content URL is invalid, sorry!"

**Screenshots**

Edition:

![Previews for Embeddable Facebook Content](http://7studio.fr/github/gutenberg/screenshot-1.png)
![Previews for Embeddable Facebook Content](http://7studio.fr/github/gutenberg/screenshot-2.png)

Website:

![Previews for Embeddable Facebook Content](http://7studio.fr/github/gutenberg/screenshot-3.png)
![Previews for Embeddable Facebook Content](http://7studio.fr/github/gutenberg/screenshot-4.png)

**Comment by danielbachhuber** [comment]
Thanks for the report, @7studio 

> Gutenberg should display different error messages for failed previews and failed oEmbed API requests.

Notably, Facebook previews don't ever work in Gutenberg (introduced in f9d17bcde8f5a119f701bac53f4498209734524c and #816). The message "Previews for this are unavailable in the editor, sorry!" is _always_ true because Gutenberg never attempts to fetch the oEmbed preview.

Flagging as `Needs Copy Review` because I think a little bit of wordsmithing could help the user understand why their Facebook embeds are behaving this way.

**Comment by danielbachhuber** [comment]
Another idea, and this would be a refinement to the existing implementation, would be to _attempt_ the oEmbed preview, and then have the preview message distinguish between a Facebook Embed that would preview on the frontend vs. one that wouldn't (because the URL is incorrect, etc.)

**Comment by michelleweber** [comment]
> why their Facebook embeds are behaving this way

Why do they -- what detail do you want to communicate here? And there's nothing the user can do about this, right?

**Comment by danielbachhuber** [comment]
> Why do they -- what detail do you want to communicate here?

Facebook Embeds don't preview in Gutenberg like, for instance, Twitter Embeds do.

> And there's nothing the user can do about this, right?

Correct.

**Comment by 7studio** [comment]
Thank you to take into account this issue @danielbachhuber 

> The message "Previews for this are unavailable in the editor, sorry!" is always true because Gutenberg never attempts to fetch the oEmbed preview.

When I used the word "preview" I wanted to say "the oEmbed content displayed" sorry for this misspelling. So in my case, the current Gutenberg message is wrong because WordPress succeeds in catching the response of oEmbed API but Gutenberg doesn't render it :wink:

I hope Facebook embeds can have the same behaviour than Twitter in the near future :grin:

**Comment by 7studio** [comment]
Not really :sweat_smile:

First of all, I didn't understand why @danielbachhuber said
> Notably, Facebook previews don't ever work in Gutenberg

I took the time to look into `core-blocks/embed` and the `Sandbox` component and now I understand why Facebook contents can't be included for the moment.

I will try to argument a solution in my other issue #8360 

IMHO, a message like:
> Unlike other embeddable content, embedded Facebook links don't preview in the editor. Take a look at the page preview to see how they look.

is the worst thing that Gutenberg can do. I don't want to switch into preview mode to know if you paste a wrong URL… Facebook contents should not be an exception.

Due to an error in my code (`/oembed/1.0/proxy/` route returned a malformatted JSON), it seems that Gutenberg is already able to display the suitable message for a wrong URL:

![Previews for Embeddable Facebook Content](http://7studio.fr/github/gutenberg/screenshot-5.png)

Gutenberg could take into account the WP_Error from the REST route response in `gutenberg_filter_oembed_result` to display the message above.

Example of WP_Error for wrong URL:
```php
WP_Error Object
(
    [errors] => Array
        (
            [oembed_invalid_url] => Array
                (
                    [0] => Not Found
                )
        )
    [error_data] => Array
        (
            [oembed_invalid_url] => Array
                (
                    [status] => 404
                )
        )
)
```
Currently, Gutenberg returns this JSON for my invalid URL:
```json
{
'html': '<a href="https://www.facebook.com/oohmyworld/vidos/2324195200940031/">https://www.facebook.com/oohmyworld/vidos/2324195200940031/</a>',
'provider_name': 'Embed Handler'
}
``` 
If Gutenberg doesn't return `html` entry, it could pass into this condition [core-blocks/embed/index.js#L207](https://github.com/WordPress/gutenberg/blob/master/core-blocks/embed/index.js#L207) and display the appropriate message :wink: 

In fact, this behaviour could be applied for all services :heart_eyes:

**Comment by danielbachhuber** [comment]
@7studio Do you want to open a pull request for the behavior you'd prefer?

**Comment by 7studio** [comment]
@danielbachhuber Yes, I will :relaxed: Thank you to take into account this issue.

**Comment by danielbachhuber** [comment]
Removing the `REST API Interaction` label because this doesn't need any changes from the REST API.
